static-assets/components/cstudio-forms/data-sources/file-browse-repo.js

```diff
@@ -64,52 +64,27 @@ YAHOO.extend(CStudioForms.Datasources.FileBrowseRepo, CStudioForms.CStudioFormDa
 				newElTitle = '';
 
 			for(var x = 0; x < datasourceDef.length; x++){
-				if (datasourceDef[x].id == this.id){
-					newElTitle = datasourceDef[x].title;
-				}
-			}
-
-			var createEl = document.createElement("div");
-			YAHOO.util.Dom.addClass(createEl, 'cstudio-form-control-node-selector-add-container-item');
-			createEl.innerHTML = CMgs.format(langBundle, "createNew") + " - " + newElTitle;
-			control.addContainerEl.create.appendChild(createEl);
-
-			var addContainerEl = control.addContainerEl;		
-			YAHOO.util.Event.on(createEl, 'click', function() {
-				control.addContainerEl = null;
-				control.containerEl.removeChild(addContainerEl);
-
-				CStudioAuthoring.Operations.uploadAsset(CStudioAuthoringContext.site, _self.processPathsForMacros(_self.repoPath), true, {
-					success: function(fileData) {
-						var item = _self.processPathsForMacros(_self.repoPath) + "/" + fileData.fileName;
-						control.insertItem(item, item, fileData.fileExtension, fileData.size, _self.id);
-						if(control._renderItems){
-							control._renderItems();
-						}
-					},
-
-					failure: function() {
-					},
-
-					context: this });
-			}, createEl);
-
-			var browseEl = document.createElement("div");
-			browseEl.innerHTML = CMgs.format(langBundle, "browseExisting") + " - " + newElTitle;
```

Comment:
Was this not needed at all?
What was it's purpose?

Reply:
@rart The createEl div shouldn't go in this datasource, since the DS is for browsing content and not uploading (that's what createEl does). When having only the browse datasourse selected (in a control) it works as expected, but when selecting multiple it renders both browse and upload entries, which shouldn't happen.